src/cljdoc/server/pedestal.clj

```diff
@@ -226,28 +235,35 @@
                  (cljdoc.render.build-log/builds-page)
                  (pu/ok-html ctx)))})
 
-(defn badge-url [status color]
-  (format "https://badgen.now.sh/badge/cljdoc/%s/%s"
-          (string/replace status  #"/" "%2F")
-          (name color)))
+(defn return-badge [ctx status color]
+  (let [url (format "https://badgen.now.sh/badge/cljdoc/%s/%s"
+                    (string/replace status  #"/" "%2F")
+                    (name color))]
+    (->> {:status 200
+          :headers {"Content-Type" "image/svg+xml;charset=utf-8"
+                    "Cache-Control" (format "public; max-age=%s" (* 30 60))}
+          :body (:body (clj-http.lite.client/get url {:headers {"User-Agent" "clj-http-lite"}}))}
+         (assoc ctx :response))))
 
 (defn badge-interceptor []
   {:name ::badge
-   :enter (fn badge [ctx]
+   :leave (fn badge [ctx]
```

Comment:
May I know what's the rationale behind implementing `:leave` instead of `:enter` for this interceptor? I just find keeping all my interceptors as the `enter`-kind easier to understand. I'd love to know any good use case for the `leave`-kind interceptors.

Reply:
Totally, great question. 👍 This interceptor also has an `:error` clause now to catch any exceptions from the resolve-version-interceptor, rendering a badge that shows a proper error message.

```clj
         :badge-for-project  [(badge-interceptor)
	                      resolve-version-interceptor
	                      (last-build-loader build-tracker)])
```
If we kept it as an `:enter` interceptor it would only be queued after the `resolve-version-interceptor` meaning the `:error` handler would also not take effect because the interceptor hasn't been ran whenever `resolve-version-interceptor` would throw an exception.

In the other places where `resolve-version-interceptor` is used the exception throwing is will result in a 500, which is the current behaviour.

Let me know if this makes sense 😄

Reply:
FWIW I'm also not super happy about mixing `:enter` and `:leave` too much but it's the best I could come up with that still works with the other `resolve-version-interceptor` use cases.

Reply:
Thanks, it totally makes sense. :)